### PR TITLE
raft: add node should reset the pendingConf state

### PR DESCRIPTION
**Pull request by absolute8511**
After add node conf proposed twice with the same node id, the pending state is not reset because
the addNode returned without setting the pending state at the second
time and the pending state will always be true unless other conf changed. During this we
can not add any new node because the propose will be ignored since the
pending state is true.

### CODE REVIEW

**Comment by xiang90** [comment]
@absolute8511 Can you please add a test case for this? Thanks!

**Comment by absolute8511** [comment]
Sure, I'll add it later.

**Comment by xiang90** [comment]
@absolute8511 Actually CI fails on the test you just added, can you fix it?

**Comment by absolute8511** [comment]
@xiang90  I fixed the test case, in the test case we should wait the previous proposal applied.

**Comment by xiang90** [comment]
Can we fix a more reliable way to do the test? Sleeping is unreliable.

File: raft/node_test.go [hunk 1]
```diff
@@ -291,6 +291,69 @@ func TestNodeProposeConfig(t *testing.T) {
...
+			case rd := <-n.Ready():
+				s.Append(rd.Entries)
+				for _, e := range rd.Entries {
+					rdyEntries = append(rdyEntries, e)
+					switch e.Type {
+					case raftpb.EntryNormal:
+					case raftpb.EntryConfChange:
+						var cc raftpb.ConfChange
+						cc.Unmarshal(e.Data)
+						n.ApplyConfChange(cc)
```
Comment (xiang90) [inline raft/node_test.go]: probably we could send a signal to a chan to notify it applied a conf change. so you can wait on that chan instead of sleeping.

Comment (absolute8511) [inline raft/node_test.go]: OK, done

**Comment by xiang90** [comment]
LGTM

**Pull request merged**